### ISSUE TITLE
Guzzle conflict

### ISSUE THREAD

**Issue by liedekef** [mollie/mollie-api-php#351]
Hi,

I'm using Mollie 2.4.1 (I saw that 2.4.2 was released 4 days ago, but I think it won't help).
I'm using this in a wordpress plugin called events-made-easy (https://wordpress.org/plugins/events-made-easy/), and have included the Mollie api by first downloading it via composer, and I include it like this:
`require_once 'payment_gateways/Mollie/2.4.1/vendor/autoload.php';`
The autoload.php is generated by composer. So I'm assuming I'm doing it ok. And it works fine too, but if another plugin that also includes Mollie is being used, I'm getting this error:
`Uncaught Error: Call to undefined function _PhpScoper5c2f536492e71\GuzzleHttp\choose_handler() in /mysite/wp-content/plugins/events-made-easy/payment_gateways/Mollie/2.4.1/vendor/guzzlehttp/guzzle/src/HandlerStack.php:37`
The other plugin is to be found at https://wordpress.org/plugins/mollie-payments-for-woocommerce/ and at first sight, they include the same api version and also an auto-generated autoload.php.
I'm even only including the Mollie API if/when needed (it seems mollie-payments-for-woocommerce is always including it on all pages).
So I'm a bit lost here. Could you provide some assistance on how to solve this?

**Comment by sandervanhooft** [comment]
Hi @liedekef,

I'm not familiar with the inner workings of WP (plugins). Did you check whether these should be compatible? It seems the error is thrown by events-made-easy, so be sure to open a ticket there.

I'll check with Mollie if there are any similar Guzzle versioning issues known.

**Comment by liedekef** [comment]
I'm the developing events-made-easy . Currently I worked around it by checking the class:
`   if (!class_exists('Mollie\Api\MollieApiClient')) {`
but it seems to me that this goes against the logic of autoloader etc ... I can't guarantee the correct (latest) Mollie API version like this.

**Comment by sandervanhooft** [comment]
I just asked around at Mollie, it seems to be related to #244 .

**Comment by davdebcom** [comment]
@sandervanhooft @liedekef This is a known issue with WordPress plugins. Also see:
https://github.com/mollie/WooCommerce/wiki/Composer-Guzzle-conflicts

Basically, WordPress doesn't support Composer so (for example) Guzzle is loading multiple times, and those are conflicting with each other.

@willemstuursma tried to solve this by using PhpScoper (which is still used), which worked in some situations but still caused issues in others. 

@sandervanhooft If you want to look into this with a fresh set of eyes, that would be awesome.

**Comment by sandervanhooft** [comment]
Thanks @davdebcom for clearing this up!

@liedekef From what I've read using PHPScoper in your package to prefix the Guzzle client libraries looks like your best chance. Unfortunately I can't help you there.

If there's any change required in this Mollie client, let me know.

**Comment by willemstuursma** [comment]
@sandervanhooft We're using Scoper to create scoped packages, see https://github.com/mollie/mollie-api-php/blob/master/Makefile.

Looks like this other module does not use a scoped version.

**Comment by willemstuursma** [comment]
@liedekef the Releases page has a scoped version as a download.

**Comment by liedekef** [comment]
And so it seems ... weird, I thought that last time I looked it wasn't there (but maybe I was just plain wrong).
A small extra question: if 2 wordpress plugins use the same scoped version, won't that cause issues too?

**Comment by Smitsel** [comment]
You might run into trouble there.

The problem is that when the first Guzzle client gets auto loaded it will have its corresponding `functions.php` loaded. This is done using a require hash in composer.

When the next client is loaded it will not automatically its `functions.php`.  The require hash is already in composer and therefore it will not load.

**Comment by Smitsel** [comment]
So i've been looking into this again, this time a little more in-depth. 
I've raised an issue with php-scoper.

If you're interested: https://github.com/humbug/php-scoper/issues/298

**Comment by sandervanhooft** [comment]
This issue is continued in the thread opened by Smitsel [here](https://github.com/humbug/php-scoper/issues/298). I'm closing this one.

Thanks @liedekef for bringing it up!

**Comment by tahsin352** [comment]
Hi All, I am just confirming that I have used 3 different WP plugins into my website at a time. It didn't get conflicted or any errors. I have used 

1. https://garazlab.com/product/wp-orangeshine-sync-wholesale-retail-fashion-clothing-apparel-accessories-boutique/
2. https://garazlab.com/product/wp-lashowroom-sync-wholesale-retail-fashion-clothing-apparel-accessories-boutique/
3. https://garazlab.com/product/wp-fashiongo-sync-wholesale-retail-fashion-clothing-apparel-accessories-boutique/

plugin. All of them uses GuzzleHttp with composer. 

I am using those successfully. 
FYI, I am using latest version of WP.
Thanks.
